src/app/util/mock/attribute-storage.cpp

```diff
@@ -0,0 +1,187 @@
+#define CLUSTER_MASK_SERVER (0x40)
+
+using namespace chip;
+using namespace chip::Test;
+using namespace chip::app;
+
+namespace {
+
+EndpointId endpoints[]    = { kMockEndpoint1, kMockEndpoint2, kMockEndpoint3 };
+uint16_t clusterIndex[]   = { 0, 2, 5 };
+uint8_t clusterCount[]    = { 2, 3, 4 };
+ClusterId clusters[]      = { MockClusterId(1), MockClusterId(2), MockClusterId(1), MockClusterId(2), MockClusterId(3),
+                         MockClusterId(1), MockClusterId(2), MockClusterId(3), MockClusterId(4) };
+uint16_t attributeIndex[] = { 0, 2, 5, 7, 11, 16, 19, 25, 27 };
+uint16_t attributeCount[] = { 2, 3, 2, 4, 5, 3, 6, 2, 2 };
+AttributeId attributes[]  = {
+    // clang-format off
+    Clusters::Globals::Attributes::ClusterRevision::Id, Clusters::Globals::Attributes::FeatureMap::Id,
+    Clusters::Globals::Attributes::ClusterRevision::Id, Clusters::Globals::Attributes::FeatureMap::Id, MockAttributeId(1),
+    Clusters::Globals::Attributes::ClusterRevision::Id, Clusters::Globals::Attributes::FeatureMap::Id,
+    Clusters::Globals::Attributes::ClusterRevision::Id, Clusters::Globals::Attributes::FeatureMap::Id, MockAttributeId(1), MockAttributeId(2),
+    Clusters::Globals::Attributes::ClusterRevision::Id, Clusters::Globals::Attributes::FeatureMap::Id, MockAttributeId(1), MockAttributeId(2), MockAttributeId(3),
+    Clusters::Globals::Attributes::ClusterRevision::Id, Clusters::Globals::Attributes::FeatureMap::Id, MockAttributeId(1),
+    Clusters::Globals::Attributes::ClusterRevision::Id, Clusters::Globals::Attributes::FeatureMap::Id, MockAttributeId(1), MockAttributeId(2), MockAttributeId(3), MockAttributeId(4),
+    Clusters::Globals::Attributes::ClusterRevision::Id, Clusters::Globals::Attributes::FeatureMap::Id,
+    Clusters::Globals::Attributes::ClusterRevision::Id, Clusters::Globals::Attributes::FeatureMap::Id
+    // clang-format on
+};
+
+} // namespace
+
+uint16_t emberAfEndpointCount(void)
+{
+    return ArraySize(endpoints);
+}
+
+uint16_t emberAfIndexFromEndpoint(chip::EndpointId endpoint)
+{
+    for (uint16_t i = 0; i < ArraySize(endpoints); i++)
+    {
+        if (endpoints[i] == endpoint)
+        {
+            return i;
+        }
+    }
+    return UINT16_MAX;
+}
+
+uint8_t emberAfClusterCount(chip::EndpointId endpoint, bool server)
+{
+    for (uint16_t i = 0; i < ArraySize(endpoints); i++)
+    {
+        if (endpoints[i] == endpoint)
+        {
+            return clusterCount[i];
+        }
+    }
+    return 0;
+}
+
+uint16_t emberAfGetServerAttributeCount(chip::EndpointId endpoint, chip::ClusterId cluster)
+{
+    uint16_t endpointIndex          = emberAfIndexFromEndpoint(endpoint);
+    uint16_t clusterCountOnEndpoint = emberAfClusterCount(endpoint, true);
+    for (uint16_t i = 0; i < clusterCountOnEndpoint; i++)
+    {
+        if (clusters[i + clusterIndex[endpointIndex]] == cluster)
+        {
+            return attributeCount[i + clusterIndex[endpointIndex]];
+        }
+    }
+    return UINT16_MAX;
```

Comment:
This does not match the non-mock impl, which returns 0 here.  Needs to match, if we're testing against the mock.